src/algoliasearch.helper.js

```diff
@@ -248,6 +250,75 @@ AlgoliaSearchHelper.prototype.searchOnce = function(options, cb) {
   });
 };
 
+/**
+ * @typedef Answer
+ * @type {object}
+ * @property {string} extract the extracted value with highlights
+ * @property {string} extractAttribute the attribute used to extract the answer
+ * @property {number} score the score indicating how well it was matched
+ */
+
+/**
+ * @typedef AnswerHit
+ * @type {object}
+ * @property {Answer} _answer the object describing why the hit was chosen
+ */
+
+/**
+ * @typedef AnswersResult
+ * @type {object}
+ * @property {AnswerHit[]} hits the answer hits
+ */
+
+ /**
+ * Start the search for answers with the parameters set in the state.
+ * This method returns a promise.
+ * @param {Object} options - the options for answers API call
+ * @param {string[]} options.attributesForPrediction - Attributes to use for predictions. If empty, `searchableAttributes` is used instead.
+ * @param {string[]} options.queryLanguages - The languages in the query. Currently only supports ['en'].
+ * @param {number} options.nbHits - Maximum number of answers to retrieve from the Answers Engine. Cannot be greater than 1000.
+ *
+ * @return {AlgoliaSearchHelper}
+ * @return {promise.<AnswersResult>} the answer results
+ */
+AlgoliaSearchHelper.prototype.findAnswers = function(options) {
+  var state = this.state;
+  var derivedHelper = this.derivedHelpers[0];
+  if (!derivedHelper) {
+    return Promise.resolve([]);
+  }
+  var derivedState = derivedHelper.getModifiedState(state);
+  var data = merge(
+    {
+      attributesForPrediction: options.attributesForPrediction,
+      nbHits: options.nbHits
+    },
+    {
+      params: omit(requestBuilder._getHitsSearchParams(derivedState), [
+        'attributesToSnippet',
+        'hitsPerPage',
+        'restrictSearchableAttributes',
+        'snippetEllipsisText' // FIXME remove this line once the engine is fixed.
+      ])
+    }
+  );
+
+  if (
+    typeof this.client.initIndex !== 'function'
+  ) {
+    throw new Error(
+      'search for answers was called, but this client does not have a function client.initIndex'
+    );
+  }
+  var index = this.client.initIndex(derivedState.index);
+  if (typeof index.findAnswers !== 'function') {
+    throw new Error(
+      'search for answers was called, but this client does not have a function client.initIndex(index).findAnswers'
```

Comment:
this is helpful, but the second error message gives pretty much the same information. Could it be consolidated to save space? How do we do it in searchForFacetValues?

Reply:
Unlike SFFV, the existence of `findAnswers` can be checked only after creating `index` here. So I couldn't merge them in a single conditional block.
https://github.com/algolia/algoliasearch-helper-js/pull/804/commits/b6d1f57923d32e198265f3dd6cdfdbc75f0ec387